### PR TITLE
DOCS: Refactor`README.md` to make it crisp and readable

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

This PR:

- Refactors development docs to development guide
- Adds a usage section that hosts the interaction, integrations and more
- Adds a contribution section to host relevant information about contributing strategy

### CODE REVIEW

**Comment by thrau** [comment]
great PR harsh! here are a few thoughts:

i appreciated that you saved all of the README content, but TBH i think we can remove all of it. we have all the content at docs.localstack.cloud now. i made sure the README content was transferred when we did the swarming.

### Installing

I know this was not part of your PR, but maybe we could add it to the scope. I think we could rephrase the story a bit. "The easiest way to install LocalStack is via pip" is misleading in my opinion, as localstack is not really being installed, it's actually the localstack-cli that's installed that then runs the docker image which contains the localstack runtime.
i don't have a concrete proposal for a phrasing, maybe you could make a suggestion?

### Running

* with respect to the "Running" section: i think we can remove all of the notes about caveats when running and just link to the getting started guide on the docs that has that information.
* i think we can remove all of the references to the SERVICES variable, which is mostly obsolete with the lazy loading of service, except for some corner cases (which again can be found in the docs) (edited)

**Comment by HarshCasper** [comment]
> also, what if we move the contributors into a subreadme into docs/community? we could extend that section later with more infos about the community

Yes, I did that first but later reverted those changes in the next commit. It is always a good idea to have contributors on the main `README.md` to show active signs of participation. But happy to do that back to ensure a more compact `README.md`

**Comment by thrau** [comment]
> > also, what if we move the contributors into a subreadme into docs/community? we could extend that section later with more infos about the community
> 
> Yes, I did that first but later reverted those changes in the next commit. It is always a good idea to have contributors on the main `README.md` to show active signs of participation. But happy to do that back to ensure a more compact `README.md`

i don't have a super strong opinion on this. i trust your judgement :-)

File: README.md [hunk 1]
```diff
@@ -61,680 +56,31 @@ The easiest way to install LocalStack is via `pip`:
...
 ```
 
-**Note**: Please do **not** use `sudo` or the `root` user - LocalStack
-should be installed and started entirely under a local non-root user. If you have problems
-with permissions in MacOS X Sierra, install with `pip install --user localstack`
+**Note**: Please do **not** use `sudo` or the `root` user - LocalStack should be installed and started entirely under a local non-root user. If you have problems with permissions in MacOS X Sierra, install with `pip install --user localstack`
 
-## Running
-
-By default, LocalStack is started inside a Docker container by running:
```
Comment (thrau) [inline README.md:70]: i think it makes sense to leave one example to start localstack via the cli, and maybe tease some of the experience that the person will have

something like
```
 % localstack start -d

     __                     _______ __             __
    / /   ____  _________ _/ / ___// /_____ ______/ /__
   / /   / __ \/ ___/ __ `/ /\__ \/ __/ __ `/ ___/ //_/
  / /___/ /_/ / /__/ /_/ / /___/ / /_/ /_/ / /__/ ,<
 /_____/\____/\___/\__,_/_//____/\__/\__,_/\___/_/|_|

 💻 LocalStack CLI 0.13.1

[20:22:20] starting LocalStack in Docker mode 🐳
[20:22:21] detaching
```
which will download and start the localstack docker container

you can query the state with:
```
 % localstack status         
┏━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Runtime version ┃ 0.13.0.10                                             ┃
│ Docker image    │ tag: latest, id: 41d3a94e8776, 📆 2021-12-05T15:37:01 │
│ Runtime status  │ ✔ running (name: "localstack_main", IP: 172.17.0.2)   │
└─────────────────┴───────────────────────────────────────────────────────┘
```
or
```
 % localstack status services
┏━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┓
┃ Service                  ┃ Status      ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━┩
│ acm                      │ ✔ available │
│ apigateway               │ ✔ available │
│ cloudformation           │ ✔ available │
│ cloudwatch               │ ✔ available │
│ config                   │ ✔ available │
│ dynamodb                 │ ✔ available │
...
```

**Review by thrau — COMMENTED** [review]
this looks much much cleaner! couple more things then i think it's ready to merge

* leave at least one example to run localstack with the CLI (see inline comment)
* i think we can remove the stargazers over time to be honest. it's not really helpful, we can maybe make it part of the community subpage
* i think we can remove all the other README pages in the docs/ folder that we already have at docs.localstack.cloud. so `docs/interaction/` `docs/developer_guides`, `docs/advanced_topics` etc..

File: README.md (hunk 1 above)
Comment (HarshCasper) [reply README.md:70]: I have created an `Example` section to exactly serve this purpose.

**Review by thrau — APPROVED** [review]
fantastic! i think we're ready to merge :-)

**Comment by whummer** [comment]
Awesome, great to see this merged @HarshCasper @thrau - so much cleaner, and now we don't need to maintain the configuration docs in two places. 👍 🎉 

On that note, looks like a few smaller changes did not make it from the README to our docs yet (e.g., `ENABLE_CONFIG_UPDATES`, `MOCK_UNIMPLEMENTED`) -  @HarshCasper  would be great if we could do a quick check on the [latest changes](https://github.com/localstack/localstack/commits/master/README.md) in the README file and add them to the [docs configuration page](https://docs.localstack.cloud/localstack/configuration). Thanks!

**Pull request merged**